contrib/guix/libexec/prelude.bash

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+export LC_ALL=C
+set -e -o pipefail
+
+# shellcheck source=contrib/shell/git-utils.bash
+source contrib/shell/git-utils.bash
+
+################
+# Required non-builtin commands should be invocable
+################
+
+check_tools() {
+    for cmd in "$@"; do
+        if ! command -v "$cmd" > /dev/null 2>&1; then
+            echo "ERR: This script requires that '$cmd' is installed and available in your \$PATH"
+            exit 1
+        fi
+    done
+}
+
+check_tools cat env readlink dirname basename git
```

Comment:
I'm a bit confused as to the policy here: Other files also appear to check for a superset of these commands. It would make sense to me if the prelude checked for tools required by the prelude (realpath, cat, git), and each script checked for the tools it needs.